src/utils/MultiInviter.js

```diff
@@ -85,18 +92,23 @@ export default class MultiInviter {
     }
 
     getErrorText(addr) {
-        return this.errorTexts[addr];
+        return this.errors[addr] ? this.errors[addr].errorText : null;
     }
 
-    async _inviteToRoom(roomId, addr) {
+    async _inviteToRoom(roomId, addr, ignoreProfile) {
         const addrType = getAddressType(addr);
 
         if (addrType === 'email') {
             return MatrixClientPeg.get().inviteByEmail(roomId, addr);
         } else if (addrType === 'mx-user-id') {
-            const profile = await MatrixClientPeg.get().getProfileInfo(addr);
-            if (!profile) {
-                return Promise.reject({errcode: "M_NOT_FOUND", error: "User does not have a profile."});
+            if (!ignoreProfile && !SettingsStore.getValue("alwaysRetryInvites", this.roomId)) {
+                const profile = await MatrixClientPeg.get().getProfileInfo(addr);
```

Comment:
This could get confusing because the exception could now be either from the profile endpoint or the invite endpoint. I would be tempted to catch any errors here and re-throw them as a single 'profile not found' error: if the remote HS is down when looking up the profile, we still want to be able to send the invite because our HS will store that for us and send it on when the remote HS comes back.